### PR TITLE
Allow yield indented object

### LINKED ISSUE #5054: Lexer: “unexpected indentation” with `yield` followed by implicit object
Possible bug or unintended behavior:

```coffee
gen = ->
    yield
      k: 'v'
```
[/#try link ](http://coffeescript.org/#try:gen%20%3D%20-%3E%0A%20%20%20%20yield%0A%20%20%20%20%20%20k%3A%20'v')

### Expected Behavior
Compile same as:
```coffee
gen = ->
    yield k: 'v'
# or
gen = ->
    yield {
        k: 'v'
    }
```

### Current Behavior
Gives compile error:
```
[stdin]:3:1: error: unexpected indentation
      k: 'v'
^^^^^^
```

### Context
I was expecting to use `yield` same as functions and statements like `return` or `throw`. This type of syntax is quite handy when returning multi key objects.

### Environment
* CoffeeScript version: 2.3.0
* Node.js version: 8.4.0

### PR DESCRIPTION
**Pull request by helixbass**
Fixes #5054 

Follows the same pattern as eg #4599 to allow `yield` to take an indented object

### CODE REVIEW

File: test/formatting.coffee [hunk 1]
```diff
@@ -417,6 +417,29 @@ test "#3199: throw multiline implicit object", ->
...
+    1
+  {value, done} = y.next()
+  ok value.type is 'a' and done is no
+
+  {value, done} = y.next()
+  ok value is 1 and done is yes
+
+  do ->
+    y =
+      if no then await
```
Comment (GeoffreyBooth) [inline test/formatting.coffee]: This needs to go in the async file, so that the tests pass in Node 6.

File: src/grammar.coffee [hunk 2]
```diff
@@ -818,6 +819,7 @@ grammar =
     o '+     Expression',                      (-> new Op '+', $2), prec: 'UNARY_MATH'
 
     o 'AWAIT Expression',                       -> new Op $1 , $2
+    o 'AWAIT INDENT Object OUTDENT',            -> new Op $1 , $3
```
Comment (GeoffreyBooth) [inline src/grammar.coffee]: Okay this is a total nit, but why is there a space before this comma? And on the previous line.

Comment (helixbass) [reply src/grammar.coffee]: @GeoffreyBooth guessing the previous line was originally copied from eg the `UNARY Expression` line, which contains an extra space to align with following. Updated

File: test/formatting.coffee (hunk 1 above)
Comment (helixbass) [reply test/formatting.coffee]: @GeoffreyBooth ok moved to `test/async.coffee`

**Comment by GeoffreyBooth** [comment]
Looks good to me. @zdenko or anyone else have any notes?

**Pull request merged**